internal/app/readycheck.go

```diff
@@ -1,13 +1,157 @@
+// Copyright 2023 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
 package app
 
-import "net/http"
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"time"
 
-func (a *Application) readinessHandler(w http.ResponseWriter, r *http.Request) {
+	"github.com/gardener/etcd-wrapper/internal/util"
 
-}
+	clientv3 "go.etcd.io/etcd/client/v3"
+	"go.uber.org/zap"
+)
+
+const (
+	// ReadyServerPort is the port number for the server that serves the readiness probe
+	ReadyServerPort       = int64(9095)
+	etcdConnectionTimeout = 5 * time.Second
+	etcdGetTimeout        = 5 * time.Second
+	etcdQueryInterval     = 2 * time.Second
+	etcdEndpointAddress   = ":2379"
+)
 
-// SetupReadinessProbe sets up the readiness probe for this application.
+// Create a struct which will hold the last status for etcd.
+// In SetupReadinessProbe first call go a.queryAndUpdateEtcdReadiness()/queryAndUpdateEtcdReadiness this function will
+// periodically query etcd and updates the readiness struct. The handler just reads from the struct.
+// If the struct has no status it assumes NotAvailable else it returns the current status.
```

Comment:
Leftover comments from planning phase? 😉

Reply:
Ah we leaked our discussion. Removed it